.github/workflows/report_jokes/main.py

```diff
@@ -0,0 +1,40 @@
+import json
+import urllib3
+
+# documentation available here : https://sv443.net/jokeapi/v2/
+JOKE_URL = 'https://v2.jokeapi.dev/joke/Programming'
+
+
+def fetch_joke() -> dict:
+    """ We fetch the joke """
+    http = urllib3.PoolManager()
+    r = http.request('GET', JOKE_URL)
+    return json.loads(r.data.decode('utf-8'))
+
+
+def validate_joke(joke: dict) -> None:
+    """ We want to validate the joke before rendering it"""
+    assert isinstance(joke, dict) and joke['error'] is False, f'Error on joke fetched... {joke}'
+
+
+def render_joke(joke: dict) -> None:
+    """ We just render on stdout the joke"""
+    validate_joke(joke)
+
+    if joke.get('type') == 'single':
+        joke_str = joke.get('joke')
+    elif joke.get('type') == 'twopart':
+        joke_str = f">> {joke.get('setup')} \n\n<< {joke.get('delivery')}"
+    else:
+        raise Exception('Unknow joke format')
+
+    print('\n🙂 Hey there, a Joke for you !')
+    print(' \\`\\`\\`')
+    print(f'{joke_str}')
+    print(' \\`\\`\\`')
+    print('😂😂😂 did you get it ?')
```

Comment:
What do you mean by `set to something` ? like it's not empty ? 🤔 
because we pass the validation up there, we're sure there is something inside that "joke_str" variable !

Reply:
If the `joke.get` raise an exception then we are good, but if it does not I supposed we could have an empty string or something unusable coming from line 25.

Reply:
ok, i get what you're saying, let's not trust the API keys...
the `.get` return a None when there is no match so force the error raising here : https://github.com/osscameroon/global-github-actions/pull/109/commits/1993c2bcd4798edf72657470bcc066c1c1c8bef3